### PR TITLE
Retrieve input/output lengths from Eager C API

### PR DESCRIPTION
**Pull request by karllessard**
Adds the equivalent of `TF_OperationInputListLength` and `TF_OperationOutputListLength` in the eager C API.

Note: I based the solution on the same algorithm found in these methods, which is not optimal in case you are having more than one input/output of interest and you need to call the API more than once for the same operation (e.g. attributes are collected and name ranges are computed on each call).

I presumed that optimizations could be applied later, maybe for both graph and eager execution mode.

CC: @alextp

### CODE REVIEW

**Review by alextp — DISMISSED** [review]

File: tensorflow/c/eager/c_api_test.cc [hunk 1]
```diff
@@ -1781,4 +1781,76 @@ TEST(CAPI, TestTFE_OpAttrsInferenceDisabledWhenNotCallingOpAddInputList) {
...
+  TFE_OpAddInputList(identityOp, inputs, 2, status);
+  CHECK_EQ(TF_OK, TF_GetCode(status)) << TF_Message(status);
+
+  // Try to retrieve lengths before executing the op (should work)
+  EXPECT_EQ(2, TFE_OpGetInputLength(identityOp, "input", status));
+  CHECK_EQ(TF_OK, TF_GetCode(status)) << TF_Message(status);
+  EXPECT_EQ(2, TFE_OpGetOutputLength(identityOp, "output", status));
+  CHECK_EQ(TF_OK, TF_GetCode(status)) << TF_Message(status);
+
+  TFE_TensorHandle* retvals[2] = { nullptr };
```
Comment (alextp) [inline tensorflow/c/eager/c_api_test.cc]: You need to call DeleteTensorHandle on retvals. Currently our internal tests are failing with a memory leak which is preventing this from getting merged

Comment (karllessard) [reply tensorflow/c/eager/c_api_test.cc]: @alextp : It's done, thanks

**Review by alextp — APPROVED** [review]

**Comment by karllessard** [comment]
Are the build failures still related to the PR?

**Comment by alextp** [comment]
@karllessard they look unrelated at first glance

**Pull request merged**